### ISSUE TITLE
Survivor Sheet (Abilities & Impairments): F&T Mastery epithet linked to one of two instances of "Mastery - Fist & Tooth"

### ISSUE THREAD

**Issue by WayForger** [toconnell/kdm-manager#506]
After updating one of my survivors to have F&T Mastery I noticed that the manager forces the survivor to have two instances of "Master - Fist & Tooth."
![screen shot 2018-07-09 at 11 18 14 pm](https://user-images.githubusercontent.com/30420007/42487352-9b7d3322-83ce-11e8-87d5-242ef01684a5.png)
![screen shot 2018-07-09 at 11 20 13 pm](https://user-images.githubusercontent.com/30420007/42487369-aed71aa0-83ce-11e8-9c7d-f73721a51be2.png)
This error remains after removing both instances of the mastery and allowing the kdm-manager site to reload. After removing only one of the two, and allowing for things to load after accepting the change, the epithet disappears. 
![screen shot 2018-07-09 at 11 18 39 pm](https://user-images.githubusercontent.com/30420007/42487543-72375a82-83cf-11e8-89be-0b74dbf242d2.png)
I also discovered that the Mastery text does not join the specialization underneath the Weapon Proficiency area in the survivor sheet overview. 
![screen shot 2018-07-09 at 11 26 29 pm](https://user-images.githubusercontent.com/30420007/42487679-0c8a26c8-83d0-11e8-861d-53b5d889b029.png)

**Comment by WayForger** [comment]
I wanted to bring this up for a little while now since first noticing it and thinking it was an odd occurrence for the survivor sheet to be throwing. I believe the Manager is having issues due to a poor interaction between the maxed out (all eight squares) Weapon Proficiency and the A&I section.

**Comment by toconnell** [comment]
@WayForger Hah! That's a good one: I think I know what might be causing it and I'll take a look at it this week.

Long story short, there's some user-friendliness code that auto-adds the mastery when the eighth box is checked, and some normalization code that attempts to update survivors when they're initialized. I think there's probably some kind of conflict happening there where the normalizer doesn't understand the max number of times a survivor can have the Mastery A&I.

Anyway, yeah, I'll take a look ASAP

**Comment by WayForger** [comment]
It's actually happening with all masteries I've found.

**Comment by toconnell** [comment]
@WayForger Yeah, it's a problem with the general logic for maxing out Hunt XP when you've got a proficiency set. 

I've got a lot going on right now and I'm not really in a position to do a new release of the Manager until August, but I'm going to try to pull this and a few other minor fixes together for a little bug-fix release that should be out...soonish.

**Comment by toconnell** [comment]
upon further investigation, it looks like we've got some legacy code that causes the mastery to be added OUTSIDE of the survivor's `add_game_asset()` method, which is...

SMH

**Comment by toconnell** [comment]
Hey @WayForger I finally got around to fighting my way through this one and the fix will go out in the release I'm working on for later this week (or potentially early this coming week).

tl;dr The issue was caused by a redundant call to the `save()` method for survivors that was getting made when the eighth point of Weapon Proficiency gets added. The extra call triggered the clean-up/normalization logic that automatically adds the mastery A&I if the survivor doesn't have it. Once that call was finished, the original call resumed and (you guessed it) finished its job to add the mastery.

At any rate, I fixed it up and cleaned up a few other aspects of adding mastery to a survivor, and things are looking nice and tidy now.

Thanks again for the write-up and the captures and the major assist on this one!
